js/piwik.js

```diff
@@ -3447,37 +3447,69 @@ if (typeof window.Matomo !== 'object') {
             }
 
             function appendAvailablePerformanceMetrics(request) {
+                var timings = '';
+
                 if (performanceAlias && performanceAlias.timing && performanceAlias
                     && performanceAlias.timing.connectEnd && performanceAlias.timing.fetchStart) {
-                    request += '&pf_net=' + (performanceAlias.timing.connectEnd - performanceAlias.timing.fetchStart);
+
+                    if (performanceAlias.timing.connectEnd < performanceAlias.timing.fetchStart) {
+                        return;
```

Comment:
@sgiehl in this case wouldn't we maybe still add other timings? Or we don't send any info as soon as one value would be negative?

Reply:
Thought about that for a while and decided to discard all values if one appears incorrect. Guess the others might not be trustworthy in that case as well.

Reply:
I guess it depends on when it actually happens. If it's because of caching or so then it might still make sense to report the others if something else is the reason for that then maybe not. I have no big preference though for now but I think ideally we should understand why it happens

Reply:
There are some notes on possible browser bugs here: https://nicj.net/navigationtiming-in-practice/
For Webkit and especially iOS there were bug reports on that: https://bugs.webkit.org/show_bug.cgi?id=188620 https://bugs.webkit.org/show_bug.cgi?id=168055 https://bugs.webkit.org/show_bug.cgi?id=186919

Reply:
👍 added a comment